AnimalTraits.java

```diff
@@ -0,0 +1,13 @@
+public abstract interface AnimalTraits{
+
+  public abstract boolean hasClaws();
```

Comment:
This is not so much an **ability** than a it is a property of an animal/creature... it would be better put in the abstract class.  Unless of course you're envisioning non-animals having this ability... but then the name of the interface wouldn't makes sense.